tests/datasets/test_datasets.py

```diff
@@ -159,6 +160,39 @@ def test_demo_dataset():
     shutil.rmtree(video_dir)
 
 
+@pytest.mark.parametrize(
```

Comment:
can you add a test for the case where we are loading a dataset to extend, but the dataset that we load has fewer trajectories than the name implies (because an error occurred during data collection before)? i think your code will handle this case correctly, but it's a tricky one, so good to test

Reply:
good thing you suggested this, there was a bug! will fix